test/extended/testdata/gssapi/fedora/base/gssapi-tests.sh

```diff
@@ -0,0 +1,305 @@
+
+#         os::cmd::expect_success_and_text "oc login -u $u -p password" 'Login failed \(401 Unauthorized\)'
+#         os::cmd::expect_failure_and_not_text 'oc whoami' $u
+#     done
+# fi
+
+# Client has uncofigured GSSAPI libs and server is GSSAPI only
+# Everything fails
+# Errors mention Kerberos
+
+if [[ "${CLIENT}" = 'CLIENT_HAS_LIBS' && "${SERVER}" = 'SERVER_GSSAPI_ONLY' ]]; then
+
+    os::test::junit::declare_suite_start "${TEST_NAME}/${CLIENT}-${SERVER}"
+
+    os::cmd::expect_failure_and_text 'oc login' 'No Kerberos credentials available'
+    os::cmd::expect_failure_and_text 'oc whoami' 'system:anonymous'
+
+    os::cmd::expect_failure_and_text 'oc login -u user1' 'An invalid name was supplied'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user1'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p wrongpassword' 'An invalid name was supplied'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p password' 'An invalid name was supplied'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_failure_and_text "oc login -u user3@${REALM}" "Can't find client principal user3@${REALM} in cache collection"
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user3'
+
+    os::cmd::expect_failure_and_text "oc login -u user4@${REALM} -p wrongpassword" "Can't find client principal user4@${REALM} in cache collection"
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user4'
+
+    os::cmd::expect_failure_and_text "oc login -u user5@${REALM} -p password" "Can't find client principal user5@${REALM} in cache collection"
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user5'
+
+    os::test::junit::declare_suite_end
+
+fi
+
+# Client has unconfigured GSSAPI libs and server is GSSAPI with Basic fallback
+# Only BASIC works
+# Errors do NOT mention Kerberos
+
+if [[ ( "${CLIENT}" = 'CLIENT_MISSING_LIBS' || "${CLIENT}" = 'CLIENT_HAS_LIBS' ) && "${SERVER}" = 'SERVER_GSSAPI_BASIC_FALLBACK' ]]; then
+
+    os::test::junit::declare_suite_start "${TEST_NAME}/${CLIENT}-${SERVER}"
+
+    os::cmd::expect_failure_and_text 'oc login <<< \n' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_text 'oc whoami' 'system:anonymous'
+
+    os::cmd::expect_failure_and_text 'oc login -u user1 <<EOF
+    EOF' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user1'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p wrongpassword' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_success_and_text 'oc login -u user2 -p password' 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' "user2@${REALM}"
+    os::cmd::expect_success_and_text 'oc logout' "user2@${REALM}"
+    os::cmd::try_until_text 'oc whoami' 'system:anonymous' # Make sure token is gone
```

Comment:
probably not necessary... the token is gone client-side after `oc logout` completes